File: OtherExercises/TasksInJavaScript/Implementation/findFirstDuplicate.js
function findFirstDuplicate(numbers) {
    let seenNumbers = new Set();
    for (let number of numbers) {
        if (seenNumbers.has(number)) {
            return number;
        }
        seenNumbers.add(number);
    }
    return -1;
}

module.exports = findFirstDuplicate;

File: OtherExercises/TasksInJava/Implementation/FindFirstDuplicate.java
import java.util.HashSet;

public class FindFirstDuplicate {
    public static int findFirstDuplicate(int[] numberArray) {
        HashSet<Integer> seenNumbers = new HashSet<>();
        for (int number : numberArray) {
            if (seenNumbers.contains(number)) {
                return number;
            }
            seenNumbers.add(number);
        }
        return -1;
    }
}
